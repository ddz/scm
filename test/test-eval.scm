;;; Test evaluator

; Basic definitions
(define a 'aye)
(define b 'bee)
(define c 'cee)

a
b
c

(set! a 'cee)
(set! c 'aye)

a
b
c

(if #t a b)
(if #f a b)
(if a b c)

; Procedures
(define list
  (lambda ls
    ls))

(define abc (list a b c))

abc

; Quasiquote
;`(a ,b)
;``(a ,(b ,b))  ; nested quasiquotes
;`(1 2 3 ,@abc) ; unquote splicing
