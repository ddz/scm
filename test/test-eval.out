> > > > aye
> bee
> cee
> > > cee
> bee
> aye
> cee
> bee
> bee
> > > (cee bee aye)
> 